File: SchoolAngular/src/app/service/student.group.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import {StudentGroup} from "../model/student.group.model";

@Injectable()
export class StudentGroupService {

  private basUrl = "http://localhost:8080/studentgroup"

  constructor(private httpClient: HttpClient) {

  }

  getStudentGroupList(): Observable<StudentGroup[]> {
    return this.httpClient.get<StudentGroup[]>(`${this.basUrl}`);
  }

  findStudentGroupListForStudent(studentId: number): Observable<StudentGroup[]> {
    return this.httpClient.get<StudentGroup[]>(`${this.basUrl}/find-for-student/${studentId}`);
  }

  createStudentGroup(studentGroup: StudentGroup): Observable<StudentGroup> {
    return this.httpClient.post<StudentGroup>(`${this.basUrl}`, studentGroup);
  }

  getStudentGroupById(id: number): Observable<StudentGroup>{
    return this.httpClient.get<StudentGroup>(`${this.basUrl}/${id}`);
  }

  updateStudentGroup(id:number, studentGroup:StudentGroup): Observable<Object>{
    return this.httpClient.put(`${this.basUrl}/${id}`, studentGroup);
  }

  deleteStudentGroup(id:number): Observable<Object>{
    return this.httpClient.delete(`${this.basUrl}/${id}`);
  }

}

File: springCurseWork/src/main/java/com/example/springcursework/servise/StudentGroupService.java
package com.example.springcursework.servise;

import com.example.springcursework.model.StudentGroup;
import java.util.List;

public interface StudentGroupService {
    public StudentGroup insert(StudentGroup groupScheduleVO);
    public List<StudentGroup> findAll();
    public void delete(int id);
    public StudentGroup findById(int id);
    public StudentGroup updateStudentGroup(int id, StudentGroup groupScheduleVO);
    public List<StudentGroup> findStudentGroupListForStudent(int studentId);
}
